### PR TITLE
Add github actions, files from repo template and prepare new release

### PR DESCRIPTION
**Pull request by gwarf**
- Add community files, GitHub Actions, drop debian, fix molecule, lint

To do:
- [X] add community files
- [X] add GitHub Actions
- [x] prepare release, bump description and version in changelog
- [x] push role to ansible galaxy when releasing
- [x] build and push container to quay when releasing

### CODE REVIEW

**Comment by gwarf** [comment]
Marking as draft while testing actions at https://github.com/gwarf/ansible-role-umd/actions

**Comment by gwarf** [comment]
GitHub actions are running at https://github.com/EGI-Federation/ansible-role-umd/actions

**Comment by gwarf** [comment]
This should now be OK and ready for review, images got built and pushed to quay from my fork.  Same for the role, but as code was taken from the master branch it's not with the correct/updated role name (umd).
We will see how it goes once this will have been merged to main.

File: .github/linters/.flake8 [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[flake8]
+# https://black.readthedocs.io/en/stable/compatible_configs.html#flake8
+extend-ignore = E203,W503
+max-line-length = 88
```
Comment (enolfc) [inline .github/linters/.flake8:4]: this seems to contradict the `.flake8` file added to the root dir

Comment (gwarf) [reply .github/linters/.flake8:4]: You are right, I dropped the one in the main directory.

Comment (gwarf) [reply .github/linters/.flake8:4]: So it's breaking the molecule GitHub action: https://github.com/gwarf/ansible-role-umd/actions/runs/3565266345/jobs/5990204325#step:4:38

I will try to find a way to configure it to use the flake conf from the action, even when running outside of GitHub Action, if you think it's OK.

Comment (gwarf) [reply .github/linters/.flake8:4]: So it works now: https://github.com/gwarf/ansible-role-umd/actions

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+          molecule_command: create
+          molecule_args: --scenario-name ${{ matrix.umd_version }}
+        env:
+          ANSIBLE_FORCE_COLOR: "1"
+
+      # TODO look into pushing some description
+      # https://github.com/marketplace/actions/update-container-description-action
+      - name: Publish image to Quay
+        run: |
+          docker commit umd-${{ matrix.os }} quay.io/egi/${{ matrix.umd_version }}:${{ matrix.os }}
```
Comment (enolfc) [inline .github/workflows/release.yml:44]: Should there be also a quay.io/egi/${{ matrix.umd_version }}:latest ?

**Review by enolfc — APPROVED** [review]
LGTM! I didn't go into all details but overall seems good.

Just a small comment on the container tags, I would also add a `latest` as this is also used in most other images.

File: .github/workflows/release.yml (hunk 2 above)
Comment (gwarf) [reply .github/workflows/release.yml:44]: In fact the repo got created to have tags corresponding to the OS release, like `umd4:centos6` or `umd4:centos7`. `umd4:centos6` is deprecated, but you get the idea, and at some point we may have `umd5:centos7` and `umd5:centos9` so I'm not sure to what should point `latest`.
We could also totally change how it's managed/organised, it's the perfect time to do this.

Comment (enolfc) [reply .github/workflows/release.yml:44]: The problem I see is:
```
$ docker pull quay.io/egi/umd4
Using default tag: latest
Error response from daemon: manifest for quay.io/egi/umd4:latest not found: manifest unknown: manifest unknown
```

we may be ok with that anyway. I would point `latest` to what we believe is the recommended image to run, for `umd4` that should pointing to `centos7`

Comment (gwarf) [reply .github/workflows/release.yml:44]: Indeed ware may be OK with this, as the usage instructions are/should be `docker pull quay.io/egi/umd4:centos7`.
That said I'm fine with having `latest` point to an arbitrary release that we recommend.
I'm not sure how to implement this, hardcoding the push to latest in the GitHub action for the specific version? What do you think/suggest?

Comment (enolfc) [reply .github/workflows/release.yml:44]: With the current structure of the job, I'm not sure how to make this simple to tag just one image as `latest` so let's skip it for the time being. Let people read documentation :)

Comment (gwarf) [reply .github/workflows/release.yml:44]: Fine, we can look into this later, if it happens that someone wants to use those images ;)

**Pull request merged**